File: src/Infraestructure/Class/Vehiculos/Camioneta.ts
import { EVehiculo } from "../../Enums/EVehiculo";
import { IVehiculo } from "../../Interfaces/IVehiculo";

export class Camioneta implements IVehiculo{
    public TipoVehiculo: EVehiculo;
    public Tarifa: number;

    public constructor(){
        this.TipoVehiculo = 2;
        this.Tarifa = 15;
    }
}

File: src/main/java/Grupo2/Garage/Infraestructure/Class/Vehiculos/Camioneta.java
package Grupo2.Garage.Infraestructure.Class.Vehiculos;

import Grupo2.Garage.Infraestructure.Enums.EVehiculo;
import Grupo2.Garage.Infraestructure.Interfaces.IVehiculo;

public class Camioneta implements IVehiculo {
    public EVehiculo TipoVehiculo;
    public Double Tarifa;
    public String Patente;

    public Camioneta(String patente){
        this.TipoVehiculo = EVehiculo.CAMIONETA;
        this.Tarifa = 15.0;
        this.Patente = patente;
    }

    @Override
    public String getTipoVehiculo() {
        return this.TipoVehiculo.name();
    }
}
